javaHapiValidatorLambda/src/main/java/software/amazon/fwoa/Handler.java

```diff
@@ -38,4 +59,51 @@ public ValidatorResponse handleRequest(String event, Context context) {
         return validate;
 
     }
+
+    private List<String> getBucketObjects(String bucketName) {
+        try {
+            List<String> objectKeys = new ArrayList<String>();
+            ListObjectsV2Request req = new ListObjectsV2Request().withBucketName(bucketName);
+            ListObjectsV2Result result;
+            do {
+                result = s3.listObjectsV2(req);
+
+                for (S3ObjectSummary objectSummary : result.getObjectSummaries()) {
+                    objectKeys.add(objectSummary.getKey());
+                }
+                // If there are more than maxKeys keys in the bucket, get a continuation token
+                // and list the next objects.
+                String token = result.getNextContinuationToken();
+                req.setContinuationToken(token);
+            } while (result.isTruncated());
+            log.info("found " + objectKeys.size() + " keys");
+            return objectKeys;
+        } catch (Exception e) {
+            throw new Error(e);
+        }
+    }
+
+    private Map<String, List<IGObject>> downloadObjects(List<String> keys, String bucketName) {
+        // keep track of .index.json objects
+        List<IGObject> indices = new ArrayList<IGObject>();
+        List<IGObject> resources = new ArrayList<IGObject>();
+        for (String key : keys) {
+            try (S3ObjectInputStream s3Object = s3.getObject(bucketName, key).getObjectContent()) {
+                IGObject bucketObj = new Utils.IGObject(key, IOUtils.toString(s3Object));
+                if (key.contains(".index.json")) {
+                    indices.add(bucketObj);
+                } else {
+                    resources.add(bucketObj);
+                }
+            } catch (Exception e) {
+                log.error(e.getMessage());
+                throw new Error(e);
```

Comment:
Do we want to rethrow original error?

Reply:
This is preferred if you want to preserve stack: https://stackoverflow.com/a/1097539 . Not sure if that's necessary as we log the error though

Reply:
Gotcha, good to know!